ui/app/js/controllers/permissions_controller.js

```diff
@@ -6,35 +6,73 @@ function($scope, $routeParams, $location, $timeout, Permissions, Search, $modal,
   $scope.loading = true;
   $scope.failed = false;
   $scope.username = $routeParams.username;
+  $scope.users = [];
+  $scope.tab = 1;
+  $scope.current_role = "All Roles";
+
+  $scope.isAllRoles = function (role) {
+    return $scope.current_role === "All Roles";
+  };
+
+  $scope.isSet = function (tabNum) {
+      return $scope.tab === tabNum;
+  };
+
+  $scope.setTab = function (newTab) {
+    $scope.tab = newTab;
+  };
+  $scope.roles_users = {};
```

Comment:
This is another place where you're duplicating data, and I think it contributes to the problem I mentioned above about newly added Roles not showing up. Instead of creating $scope.users and $scope.roles_users, you can use one data structure that holds them both. You'll need to update the templates a little bit to cope with this.